controller/lab_case_controller.py

```diff
@@ -13,16 +14,48 @@ def register_lab_case(case: LabCase):
 class LabCaseControllerImpl(LabCaseController):
     def __init__(self, db: LabCaseDB):
         self.db = db
-        
           
     def register_lab_case(self, case: LabCase):
         if self.db.fetch(case.case_id) == None:
             self.db.save(case)
         else:
             self.db.update(case)
+    
+    def import_allele_table(self, case: LabCase, file: str):
+        file = "tests/assets/allele_table_example.csv"
+        
+        with open(file) as csv_file:
+            csv_reader = csv.reader(csv_file, delimiter=',')
+            lines = []
+            for row in csv_reader:
+                lines.append(row)
+        
+        for line in lines:
+            if len(line) == 5:
+                line.pop()
+            for i, v in enumerate(line):
+                line[i] = line[i].strip()
+            if line[3] == "":
+                line[3] = line[2]
+        
+        for line in lines[1:]:
+            name = line[0]
+            genotype = Genotype(locus = line[1], allele_1 = line[2], allele_2 = line[3])
+            
+            subject = self.__get_subject(case, name)
+            
+            if subject != None:
+                subject.genotype.append(genotype)
+            else:
+                case.subjects.append(Subject(name, [genotype]))                    
             
-    # def fetch(self, case: int) -> LabCase:
-    # for saved_case in self.lab_cases:
-    #     if(saved_case.case_id == case):
-    #         return saved_case
-    # return None
\ No newline at end of file
+    def __get_subject(self, case, name):
+        for subject in case.subjects:
+            if subject.name == name:
+                return subject
+        return None
+            
+        # subject type
+        # https://regexr.com/
+        
+    # file.close()
```

Comment:
Do we need to call `file.close()` when we finish using a file that we've opened? If so, let's call it in `import_allele_table`